nexus/operation-ids.md

```diff
@@ -0,0 +1,172 @@
+operation. These asynchronous operations can be cancelled and checked for their outcomes and current state. The caller
+can also specify a callback URL, which the handler uses to asynchronously deliver the result of an operation's when it
+is ready.
+
+## Operation identifiers
+
+Since an asynchronous operation is modelled on top of synchronous RPCs, it must be addressable (identifiable) by both
+the caller and the handler. There are two approaches for generating identifiers, caller side and handler side.
+
+### Caller generated identifiers
+
+With this approach, the caller provides an Id to each of the pre-defined RPCs.
+
+The handler _should_ qualify an identifier when dealing with untrusted sources and multi-tenant applications.
+
+The following set of RPCs would be needed to model asynchronous operations using client generated Ids:
+
+- `Start(operation, id, input, callbackURL)` - handler responds inline or indicates that an async operation was started
+- `Cancel(operation, id)` - cancel operation by Id
+- `GetResult(operation, id)` - get operation result by Id
+- `Describe(operation, id)` - get operation status by Id
+- `DeliverResult(token)` - deliver a result of an async operation using a token in the Start request's callback URL
+
+Example invocations:
+
+```js
+Start(
+  'charge',
+  transactionId,
+  { customerId: 'someCustomerId', amount: 5000 },
+  'nexus://caller.example.com/callback?token=' + token
+);
+Cancel('charge', transactionId);
+```
+
+#### Handler addressability
+
+In some cases, the handler might require more than just a single identifier for a given operation, e.g. when an
+operation is backed by a workflow update, which requires both a workflow Id and an update Id. In order to avoid
+maintaining a handler side persistent mapping between caller and handler identifiers, the handler may choose to "leak"
+this detail to the caller, which would need to concatinate multiple identifiers for a given call. The example above
+becomes:
+
+```js
+Start(
+  'charge',
+  customerId + '/' + transactionId,
+  { amount: 5000 },
+  'nexus://caller.example.com/callback?token=' + token
+);
+Cancel('charge', customerId + '/' + transactionId);
+```
+
+> NOTE: This looks a lot like a nested REST style URL, e.g: `/customers/{customerId}/charges/{transactionId}`.
+
+> To take this approach to the extreme, `operation` and `id` could be combined into a single `path` argument, as shown
+> below:
+>
+> - `Start(path, input, callbackURL)`
+> - `Cancel(path)`
+> - `GetResult(path)`
+> - `Describe(path)`
+
+##### Mapping from client to handler identifiers
+
+To avoid leaking the identifier construction logic to the client, the handler would need a mapping of client identifiers
+to handler identifiers. Temporal can provide a strongly consistent store out of the box for this mapping.
+
+A single execution would need to map to multiple client identifiers to account for workflow starts and updates.
```

Comment:
```suggestion
A single workflow execution would need to map to multiple client identifiers to account for workflow starts and updates.
```

to differentiate from operation execution

Reply:
This is actually operation execution, I'll clarify